src/search/algorithms/equivalence_relation.h

```diff
@@ -1,20 +1,23 @@
 #ifndef ALGORITHMS_EQUIVALENCE_RELATION_H
 #define ALGORITHMS_EQUIVALENCE_RELATION_H
 
+#include "../utils/system.h"
+
 #include <algorithm>
-#include <cmath>
 #include <list>
 #include <unordered_map>
 #include <vector>
 
 namespace equivalence_relation {
 class Block;
-typedef std::list<int>::iterator ElementListIter;
-typedef std::list<int>::const_iterator ElementListConstIter;
-typedef std::list<Block>::iterator BlockListIter;
-typedef std::list<Block>::const_iterator BlockListConstIter;
 class EquivalenceRelation;
 
+using ElementListIter = std::list<int>::iterator;
+using ElementListConstIter = std::list<int>::const_iterator;
+using BlockListIter = typename std::list<Block>::iterator;
+using BlockListConstIter = typename std::list<Block>::const_iterator;
```

Comment:
I think it is because the type `const_iterator` is nested in `std::list<Block>`.

Reply:
I don't think that would explain why is it not needed for the first two definitions, which are the same as the last two with `int` instead of `Block`.

Reply:
I think this is a left-over of the intermediate stage where we actually templated the whole class. I assume it can go (again).